src/Extension/DefaultExtension.php

```diff
@@ -605,6 +605,98 @@ public function smarty_modifier_join($values, $separator = '')
 		return implode((string) ($separator ?? ''), (array) $values);
 	}
 
+	/**
+	 * Smarty json_encode modifier plugin.
+	 * Type:     modifier
+	 * Name:     json_encode
+	 * Purpose:  Returns the JSON representation of the given value or false on error. The resulting string will be UTF-8 encoded.
+	 *
+	 * @param mixed $value
+	 * @param int   $flags
+	 * @param string $input_encoding of $value; defaults to \Smarty\Smarty::$_CHARSET
+	 *
+	 * @return string|false
+	 */
+	public function smarty_modifier_json_encode($value, $flags = 0, string $input_encoding = null)
+	{
+		if (!$input_encoding) {
+			$input_encoding = \Smarty\Smarty::$_CHARSET;
+		}
+
+		# json_encode() expects UTF-8 input, so recursively encode $value if necessary into UTF-8
+		if ($value && strcasecmp($input_encoding, 'UTF-8')) {
+			if (is_string($value)) {	# shortcut for the most common case
+				$value = mb_convert_encoding($value, 'UTF-8', $input_encoding);
+			}
+			elseif (is_array($value) || is_object($value)) {
+				static $transcoder;	# this closure will be assigned once, and then persist in memory
+				if (is_null($transcoder)) {
+					/**
+					* Similar to mb_convert_encoding(), but operates on keys and values of arrays, and on objects too.
+					* Objects implementing \JsonSerializable and unsupported types are returned unchanged.
+					*
+					* @param string $from_encoding
+					* @param string $to_encoding
+					* @param mixed $data
+					* @return mixed
+					*/
+					$transcoder = function($data, string $to_encoding, string $from_encoding) use(&$transcoder) {
+						if (empty($data)) {
+							return $data;
+						}
+						elseif (is_string($data)) {
+							return mb_convert_encoding($data, $to_encoding, $from_encoding);
+						}
+						elseif (is_scalar($data)) {
+							return $data;
+						}
+
+						# convert object to array if necessary
+						if (is_object($data)) {
+							if (is_a($data, '\JsonSerializable')) {	# this is the only reason why this function is not generic
```

Comment:
```suggestion
							if (is_a($data, \JsonSerializable::class)) {	# this is the only reason why this function is not generic
```

Reply:
Looks good too, especially if that helps IDEs check code.